pkg/webhook/cert/provisioner.go

```diff
@@ -32,62 +33,92 @@ import (
 // destination - such as a Secret or local file. Provisioner can update the CA field of
 // certain resources with the CA of the certs.
 type Provisioner struct {
-	Client client.Client
 	// CertGenerator generates certificate for a given common name.
 	CertGenerator generator.CertGenerator
 	// CertWriter knows how to persist the certificate.
 	CertWriter writer.CertWriter
 
+	// ClientConfig is the WebhookClientCert that contains the information to generate
+	// the certificate. The CABundle will be injected to the ClientConfig.
+	// The ClientConfig will be used to inject into other objects,
+	// e.g. MutatingWebhookConfiguration and ValidatingWebhookConfiguration.
+	ClientConfig *admissionregistrationv1beta1.WebhookClientConfig
+
 	once sync.Once
 }
 
-// Sync takes a runtime.Object which is expected to be either a MutatingWebhookConfiguration or
-// a ValidatingWebhookConfiguration.
-// It provisions certificate for each webhook in the webhookConfiguration, ensures the cert and CA are valid,
-// and not expiring. It updates the CABundle in the webhook configuration if necessary.
-func (cp *Provisioner) Sync(webhookConfiguration runtime.Object) error {
-	var err error
+// Provision provisions certificates for for the WebhookClientConfig.
+// It ensures the cert and CA are valid and not expiring.
+// It updates the CABundle in the webhookClientConfig if necessary.
+func (cp *Provisioner) Provision(dryrun bool) (bool, error) {
```

Comment:
I am wondering if we should combine `Provision` and `Inject` methods into one by introducing parameters to `Provision` method because ideally, that should be one operation. provision and inject cc in the objects ?